pkg/controllers/management/globaldns/globaldns_handler.go

```diff
@@ -196,20 +196,28 @@ func (n *GDController) generateNewIngressSpec(globaldns *v3.GlobalDNS) *v1beta1.
 	}
 }
 
-func (n *GDController) updateIngressEndpoints(ingress *v1beta1.Ingress, endpoints []string) error {
+func (n *GDController) updateIngressEndpointsFQDN(ingress *v1beta1.Ingress, obj *v3.GlobalDNS) error {
+	var err error
 
-	if !n.ifEndpointsDiffer(ingress.Status.LoadBalancer.Ingress, endpoints) {
-		return nil
-	}
+	if n.ifEndpointsDiffer(ingress.Status.LoadBalancer.Ingress, obj.Status.Endpoints) {
+		ingress.Status.LoadBalancer.Ingress = n.sliceToStatus(obj.Status.Endpoints)
+		ingress, err = n.ingresses.UpdateStatus(ingress)
 
-	ingress.Status.LoadBalancer.Ingress = n.sliceToStatus(endpoints)
-	updatedObj, err := n.ingresses.UpdateStatus(ingress)
+		if err != nil {
+			return fmt.Errorf("GlobalDNSController: Error updating Ingress %v", err)
+		}
+	}
 
-	if err != nil {
-		return fmt.Errorf("GlobalDNSController: Error updating Ingress %v", err)
+	if len(ingress.Spec.Rules) > 0 {
```

Comment:
when would len of ingress.spec.Rules be 0? and do we have to handle it?

Reply:
We create the ingress here https://github.com/rancher/rancher/blob/b78355467e57d8103a20a25bbe455dc60ef4b6b4/pkg/controllers/management/globaldns/globaldns_handler.go#L90 and set the ingress.spec.Rules at that time and the hostname as well. 

This is method where we make the check is called when the ingress is already existing, only for updates. So  ingress.Spec.Rules won't be empty really, but adding the check for validation and index checks.